client/indexer-service/src/handler.rs

```diff
@@ -332,6 +342,16 @@ impl IndexerService {
                 multiaddresses,
                 capacity,
             } => {
+                let current_block_hash = self.client.info().best_hash;
+
+                let stake = self
+                    .client
+                    .runtime_api()
+                    .get_bsp_stake(current_block_hash, bsp_id)
+                    .expect("to have a stake")
```

Comment:
Up for discussion whether or not we want to leave the possibility of a panic here 🤔

Reply:
Keep in mind that any error here will result in the retry to index on the next finality notification.

While I agree with "if something really bad and unrecoverable happened, let's abort" I'm not sure if there is any case where the result may be different later (in which case would be recoverable).

Reply:
Agreed, if this happened I wouldn't even know why it is happening. It would be an error in the runtime even. So panic is fine. This would be unrecoverable.

Reply:
The other option is to skip the update of the stake value and we just have an error message. But the database would be inconsistent with what is really happening in chain.

Or we retry until we have an answer which could block the indexing process if the issue cannot be solved on its own.